source/whats-new.txt

```diff
@@ -82,6 +82,76 @@ The 3.0 driver release includes the following new features:
   `TimeOnly Struct. <https://learn.microsoft.com/en-us/dotnet/api/system.timeonly?view=net-6.0>`__
   API reference page on MSDN.
 
+- Adds support for client-side projection when using the ``Find()`` method,
+  the ``Select()`` method, or the ``Project()`` aggregation stage.
+  In previous versions of the driver, you could perform client-side
+  projection only after calling the ``ToEnumerable()`` or ``AsEnumerable()`` method, and
+  only within the ``Select()`` method.
+  
+  Select a tab below to see how to enable and use client-side projection for the
+  corresponding method.
+
+  .. tabs::
+
+    .. tab:: Find() Method
+       :tabid: find-method
+
+       .. code-block:: csharp
+
+          // Enable client-side projection
+          var findOptions = new FindOptions;
+          findOptions.TranslationOptions = new ExpressionTranslationOptions
+          {
+              EnableClientSideProjections = true
+          };
+         
+          var find = collection
+              .Find(doc => doc.Id == 1, findOptions);
+              .Project(doc => new { R = MyFunction(doc.Name) });
+  
+    .. tab:: Select() Method
+       :tabid: select-method
+
+       .. code-block:: csharp
+
+          // Enable client-side projection
+          var translationOptions = new ExpressionTranslationOptions
+          {
+              EnableClientSideProjections = true
+          };
+         
+          var queryable = collection
+              .AsQueryable(translationOptions)
+              .Where(doc => doc.Id == 1)
+              .Select(doc => new { R = MyFunction(doc.Name) });
+
+    .. tab:: Project() Method
+       :tabid: project-method
+
+       .. code-block:: csharp
+
+          // Enable client-side projection
+          var aggregateOptions = new AggregateOptions;
+          aggregateOptions.TranslationOptions = new ExpressionTranslationOptions
+          {
+              EnableClientSideProjections = true
+          };
+
+          var aggregate = collection
+              .Aggregate(pipeline, aggregateOptions)
```

Comment:
Some of your tests call `Aggregate()` with no arguments, but I couldn't find that method overload and wasn't sure whether it accepted an `options` argument--hence the `pipeline` argument.

Reply:
The correct syntax is:

```collection.Aggregate(aggregateOptions)```

There is no overload that takes a `pipeline`.

`Aggregate` is an *extension* method:

``` public static IAggregateFluent<TDocument> Aggregate<TDocument>(this IMongoCollection<TDocument> collection, AggregateOptions options = null)```

and the `options` argument is optional so:

```collection.Aggregate()```

is the same as:

```collection.Aggregate(options: null)```

Reply:
Maybe I confused it with [Aggregate<TResult>](https://mongodb.github.io/mongo-csharp-driver/2.29.0/api/MongoDB.Driver/MongoDB.Driver.IMongoCollection-1.Aggregate.html)